### ISSUE TITLE
Add multi-valued attributes

### ISSUE THREAD

**Issue by ajcarpenter42** [thomaspoignant/scim-patch#38]
**Describe the bug**
"Add" operation does not function correctly for multi-valued attributes. Per https://tools.ietf.org/html/rfc7644#page-33, an add operation for a multi-valued attribute should use an array value, i.e. `{ "op":"add", "path": "members", "value": [ { ... } ] }`, but scimPatch treats the array as the value, as if it expects `{ "op":"add", "path": "members", "value": { ... } }`.

**To Reproduce**
(Javascript)
```
const scimPatch = require("scim-patch");
const assert = require('assert');

const scimUser = {
  schemas: ['urn:ietf:params:scim:schemas:core:2.0:User'],
  members: [
    {
      display: "The Wizzard",
      "$ref": "https://example.com/v2/Users/xxxx",
      value: "xxxx"
    },
    // Should also be able to comment this member out...
    {
      display: "Babs Jensen",
      "$ref": "https://example.com/v2/Users/2819c223-7f76-453a-919d-413861904646",
      value: "2819c223-7f76-453a-919d-413861904646"
    }
    // ... and get the same result
  ]
};
const patches = [
  {
    op: 'Add',
    path: 'members',
    value: [
      {
        display: "Babs Jensen",
        "$ref": "https://example.com/v2/Users/2819c223-7f76-453a-919d-413861904646",
        value: "2819c223-7f76-453a-919d-413861904646"
      }
    ]
  }
];


try {
  scimPatch.patchBodyValidation({
    schemas: ['urn:ietf:params:scim:api:messages:2.0:PatchOp'],
    Operations: patches
  });
}
catch (error) {
  console.log(error);
}

const patchedScimUser = (()=>{
  try {
    return scimPatch.scimPatch(scimUser, patches);
  }
  catch (error) {
    console.log(error);
    return scimUser;
  }
})();

const expectMembers = [
  {
    display: "The Wizzard",
    "$ref": "https://example.com/v2/Users/xxxx",
    value: "xxxx"
  },
  {
    display: "Babs Jensen",
    "$ref": "https://example.com/v2/Users/2819c223-7f76-453a-919d-413861904646",
    value: "2819c223-7f76-453a-919d-413861904646"
  },
];

patchedScimUser.members.forEach(
  (member, i) => assert.deepEqual(
    expectMembers.length > i ? expectMembers[i] : undefined,
    member,
  )
);
```

**Expected behavior**
Assertion should succeed. The new member should be found to exist in the "members" array and should not be added. (If the member were not found, it should be added.)

**Comment by ajcarpenter42** [comment]
Apologies if this is an amateur question, but the contribution guide seems to imply that you'll accept contributions from other users. However, when I attempted to create a branch for a pull request, and I attempted to push, I was forbidden.

```
$ git push --set-upstream origin 'fix/#38-add-array-to-array'
Username for 'https://github.com': ajcarpenter42
Password for 'https://ajcarpenter42@github.com':
remote: Permission to thomaspoignant/scim-patch.git denied to ajcarpenter42.
fatal: unable to access 'https://github.com/thomaspoignant/scim-patch.git/': The requested URL returned error: 403
```

**Comment by thomaspoignant** [comment]
> Apologies if this is an amateur question, but the contribution guide seems to imply that you'll accept contributions from other users. However, when I attempted to create a branch for a pull request, and I attempted to push, I was forbidden.
> 
> ```
> $ git push --set-upstream origin 'fix/#38-add-array-to-array'
> Username for 'https://github.com': ajcarpenter42
> Password for 'https://ajcarpenter42@github.com':
> remote: Permission to thomaspoignant/scim-patch.git denied to ajcarpenter42.
> fatal: unable to access 'https://github.com/thomaspoignant/scim-patch.git/': The requested URL returned error: 403
> ```

@ajcarpenter42 you can fork the repo and open a pull request.
You can follow this tutorial to do it: https://jarv.is/notes/how-to-pull-request-fork-github/
Even if @alicial was faster this time 🥇

**Comment by thomaspoignant** [comment]
@ajcarpenter42 the fix is available in the `v0.2.7` version of the library.
https://www.npmjs.com/package/scim-patch/v/0.2.7
